Scripts/BuildEngine.ps1

```diff
@@ -24,151 +24,160 @@
 #Requires -PSEdition Core
 
 param (
-    [Parameter(HelpMessage="System name to build, default to all")]
-    [ValidateSet('Windows', 'Linux', 'Darwin')]
-    [string[]] $SystemNames = @('Windows', 'Linux', 'Darwin'),
-
-    [Parameter(HelpMessage="Architecture type to build, default to x64")]
-    [ValidateSet('x64')]
-    [string] $Architectures = 'x64',
-
     [Parameter(HelpMessage="Configuration type to build")]
     [ValidateSet('Debug', 'Release')]
     [string[]] $Configurations = @('Debug', 'Release'),
 
     [Parameter(HelpMessage="Whether to run build, default to True")]
-    [bool] $RunBuilds = $True
+    [bool] $RunBuilds = $True,
+
+    [Parameter(HelpMessage="VS version use to build, default to 2019")]
+    [ValidateSet(2019, 2022)]
+    [int] $VSVersion = 2019
 )
 
+
+##Define None variables
+Set-Variable ARCHITECTURE -Option None -value 'x64'
+Set-Variable SYSTEM_NAMES -Option None -value 'Windows', 'Linux', 'Darwin'
+
+# # CMake arguments variables
+
+Set-Variable CMAKE_GENERATOR_DARWIN -Option None -value 'Xcode'
+Set-Variable CMAKE_GENERATOR_UNIX -Option None -value 'Unix Makefiles'
+
+# VS Version supported
+Set-Variable VS16 -Option None -value 'Visual Studio 16 2019'
+Set-Variable VS17 -Option None -value 'Visual Studio 17 2022'
+
+
 $ErrorActionPreference = "Stop"
 [string]$RepoRoot = [IO.Path]::Combine($PSScriptRoot, "..")
 
 . (Join-Path $PSScriptRoot Shared.ps1)
 
 $CMakeProgram  =  Find-CMake
+
 if($CMakeProgram) {
     Write-Host "CMake program found..."
+}else{
+    throw 'CMake program not found'
+    exit(1)
 }
 
-function Build([string]$systemName, [string]$architecture, [string]$configuration, [bool]$runBuild) {
-    $IsMultipleConfig = ($systemName -eq 'Windows')
+function Build([string]$Configuration ="Debug", [int]$VSVersion = 2019 ,[bool]$RunBuild =$True) {
+    
+    #Check if the OS allow multiple configguration, if not windows this value will'be false
+    $IsMultipleConfig = $IsWindows
+
+    #Check System name
+    if ($IsLinux) {
+        $SystemName =  "Linux"
+    }
+    elseif ($IsMacOS) {
+        $SystemName =   "Darwin"
+    }
+    elseif ($IsWindows) {
+        $SystemName =  "Windows"
+    }else{
+        throw 'The OS is not supported' 
+    }
+
 
-    Write-Host "Building $systemName $architecture $configuration"
+    Write-Host "Building $SystemName $Configuration"
```

Comment:
You should add the `$ARCHITECTURE`, the dev we know for which  architecture he/her is building against

Reply:
ok